pkg/build/strategies/dockerfile/dockerfile.go

```diff
@@ -0,0 +1,355 @@
+func (builder *Dockerfile) CreateDockerfile(config *api.Config) error {
+	glog.V(4).Infof("Constructing image build context directory at %s", config.WorkingDir)
+	buffer := bytes.Buffer{}
+
+	if len(config.ImageWorkDir) == 0 {
+		config.ImageWorkDir = "/opt/app-root/src"
+	}
+
+	imageUser := config.AssembleUser
+
+	// where files will land inside the new image.
+	scriptsDestDir := filepath.Join(getDestination(config), "scripts")
+	sourceDestDir := filepath.Join(getDestination(config), "src")
+
+	buffer.WriteString(fmt.Sprintf("FROM %s\n", config.BuilderImage))
+
+	if len(config.AssembleUser) > 0 {
+		buffer.WriteString(fmt.Sprintf("USER %s\n", imageUser))
+	}
+
+	generatedLabels := util.GenerateOutputImageLabels(builder.sourceInfo, config)
+	if len(generatedLabels) > 0 || len(config.Labels) > 0 {
+		first := true
+		buffer.WriteString("LABEL ")
+		for k, v := range generatedLabels {
+			if !first {
+				buffer.WriteString(fmt.Sprintf(" \\\n      "))
+			}
+			buffer.WriteString(fmt.Sprintf("%q=%q", k, v))
+			first = false
+		}
+		for k, v := range config.Labels {
+			if !first {
+				buffer.WriteString(fmt.Sprintf(" \\\n      "))
+			}
+			buffer.WriteString(fmt.Sprintf("%q=%q", k, v))
+			first = false
+		}
+
+		buffer.WriteString("\n")
+	}
+
+	// only COPY scripts dir if required scripts are present, i.e. the dir is not empty;
+	// even if the "scripts" dir exists, the COPY would fail if it was empty
+	scriptsProvided, fileNames := checkValidDirWithContents(filepath.Join(config.WorkingDir, builder.uploadScriptsDir))
+	assembleProvided := false
+	runProvided := false
+	for _, f := range fileNames {
+		glog.V(2).Infof("found override script file %s", f.Name())
+		if f.Name() == "run" {
+			runProvided = true
+		} else if f.Name() == "assemble" {
+			assembleProvided = true
+		}
+		if runProvided && assembleProvided {
+			break
+		}
+	}
+
+	env := createBuildEnvironment(config.WorkingDir, config.Environment)
+
+	buffer.WriteString(fmt.Sprintf("%s", env))
+	if scriptsProvided {
+		glog.V(2).Infof("Override scripts are included in directory %q", builder.uploadScriptsDir)
+		buffer.WriteString("# Copying in override assemble/run scripts\n")
+		buffer.WriteString(fmt.Sprintf("COPY --chown=%s:0 %s %s\n", sanitize(imageUser), sanitize(filepath.ToSlash(builder.uploadScriptsDir)), sanitize(filepath.ToSlash(scriptsDestDir))))
+	}
+
+	// copy in the user's source code.
+	buffer.WriteString("# Copying in source code\n")
+	buffer.WriteString(fmt.Sprintf("COPY --chown=%s:0 %s %s\n", sanitize(imageUser), sanitize(filepath.ToSlash(builder.uploadSrcDir)), sanitize(filepath.ToSlash(sourceDestDir))))
+
+	glog.V(2).Infof("Processing secret inputs: %#v", config.Injections)
+	config.Injections = util.FixInjectionsWithRelativePath(config.ImageWorkDir, config.Injections)
+	glog.V(2).Infof("Processed secret inputs: %#v", config.Injections)
+
+	if len(config.Injections) > 0 {
+		buffer.WriteString("# Copying in injected secret/configmap content\n")
```

Comment:
rm "secret/configmap" - things besides k8s secrets and configMaps can be injected, and the injections API is agnostic w.r.t. the source.

Reply:
will update